File: ruby/batching_urls_sample.rb
require 'openssl'
require 'base64'
require 'cgi'
require 'json'
require 'net/http'
require 'uri'

# You can obtain you access id and secret key here: http://moz.com/products/api/keys
ACCESS_ID	= "ACCESS_ID_HERE"
SECRET_KEY	= "SECRET_KEY_HERE"

# Set your expires for several minutes into the future.
# Values excessively far in the future will not be honored by the Mozscape API.
expires	= Time.now.to_i + 300

# A new linefeed is necessary between your AccessID and Expires.
string_to_sign = "#{ACCESS_ID}\n#{expires}"

# Get the "raw" or binary output of the hmac hash.
binary_signature = OpenSSL::HMAC.digest('sha1', SECRET_KEY, string_to_sign)

# We need to base64-encode it and then url-encode that.
URL_SAFE_SIGNATURE = CGI::escape(Base64.encode64(binary_signature).chomp)

# This is the URL that we want metrics for.
object_url = 'www.seomoz.org'

# Add up all the bit flags you want returned.
# Learn more here: http://apiwiki.moz.com/query-parameters/
cols = '103079215108'

# Now put your entire request together.
# This example uses the Mozscape URL Metrics API.
request_url = "http://lsapi.seomoz.com/linkscape/url-metrics/?Cols=#{cols}&AccessID=#{ACCESS_ID}&Expires=#{expires}&Signature=#{URL_SAFE_SIGNATURE}"

# Put your URLS into an array and json_encode them.
batched_domains = ['www.moz.com', 'www.apple.com', 'www.pizza.com']
encoded_domains = batched_domains.to_json

# Go and fetch the URL
uri = URI.parse("#{request_url}")
http = Net::HTTP.new(uri.host, uri.port)
request = Net::HTTP::Post.new(uri.request_uri)
request.body = encoded_domains
response = http.request(request)

puts response.body

File: php/batching_urls_sample.php
<?php
// Get your access id and secret key here: https://moz.com/products/api/keys
$accessID = "ACCESS_ID_HERE";
$secretKey = "SECRET_KEY_HERE";

// Set your expires times for several minutes into the future.
// An expires time excessively far in the future will not be honored by the Mozscape API.
$expires = time() + 300;

// Put each parameter on a new line.
$stringToSign = $accessID."\n".$expires;

// Get the "raw" or binary output of the hmac hash.
$binarySignature = hash_hmac('sha1', $stringToSign, $secretKey, true);

// Base64-encode it and then url-encode that.
$urlSafeSignature = urlencode(base64_encode($binarySignature));

// Add up all the bit flags you want returned.
// Learn more here: https://moz.com/help/guides/moz-api/mozscape/api-reference/url-metrics
$cols = "68719476736";

// Put it all together and you get your request URL.
$requestUrl = "http://lsapi.seomoz.com/linkscape/url-metrics/?Cols=".$cols."&AccessID=".$accessID."&Expires=".$expires."&Signature=".$urlSafeSignature;

// Put your URLS into an array and json_encode them.
$batchedDomains = array('www.moz.com', 'www.apple.com', 'www.pizza.com');
$encodedDomains = json_encode($batchedDomains);

// Use Curl to send off your request.
// Send your encoded list of domains through Curl's POSTFIELDS.
$options = array(
	CURLOPT_RETURNTRANSFER => true,
	CURLOPT_POSTFIELDS     => $encodedDomains
	);

$ch = curl_init($requestUrl);
curl_setopt_array($ch, $options);
$content = curl_exec($ch);
curl_close( $ch );

$contents = json_decode($content);
print_r($contents);
?>
